### ISSUE TITLE
BUG: incorrect warning message when call `np.linalg.det` on apple M1

### ISSUE THREAD

**Issue by husisy** [numpy/numpy#24633]
### Describe the issue:

On Apple M1, calling `np.linalg.det` on complex matrix generates incorrect warning message (still correct results).

| env | libblas | results | warning |
| :-: | :-: | :-: | :-: |
| `env00` | openblas | correct | yes |
| `env01` | accelerate | correct | no |

minimum working example (MWE) to reproduce warning

```bash
#libblas from openblas
conda create -c conda-forge -n env00 numpy
conda activate env00
```

MWE without warning (use `libblas` from `accelerate`)

```bash
conda create -c conda-forge -n env01 numpy "libblas=*=*accelerate"
conda activate env01
```

seems related issues: numpy/numpy#22025

I also did other tests

1. no warning message on `ubuntu-22.04,AMD R7,openblas,dtype=complex128`
2. no warning message on `mac-M1, openblas, dtype=float64`

### Reproduce the code example:

```python
# python -c "import numpy as np; print(np.linalg.det(np.array([[0,1],[1,0]], dtype=np.complex128)))"
import numpy as np
print(np.linalg.det(np.array([[0,1],[1,0]], dtype=np.complex128)))
# (-1+0j) #still give correct results
```

### Error message:

```shell
/Users/zhangc/miniconda3/envs/env00/lib/python3.11/site-packages/numpy/linalg/linalg.py:2180: RuntimeWarning: divide by zero encountered in det
  r = _umath_linalg.det(a, signature=signature)
/Users/zhangc/miniconda3/envs/env00/lib/python3.11/site-packages/numpy/linalg/linalg.py:2180: RuntimeWarning: invalid value encountered in det
  r = _umath_linalg.det(a, signature=signature)
```

### Runtime information:

```Python
import sys, numpy
print(numpy.__version__); print(sys.version)
# 1.25.2
# 3.11.5 | packaged by conda-forge | (main, Aug 27 2023, 03:33:12) [Clang 15.0.7 ]
```

### Context for the issue:

Low priority, it gives the correct results.

**Comment by seberg** [comment]
Thanks for the report.  I cannot reproduce this with our 1.26 wheels, so I suspect this is an OpenBLAS issue that was already fixed. (Using OpenBLAS version `0.3.23.dev`).

**Comment by seberg** [comment]
Ah, I can however reproduce it with the 1.25.2 wheels, so I think I will close this as fixed in OpenBLAS, thanks again.

**Comment by JMuff22** [comment]
Fixed this issue with `conda install -c conda-forge numpy "libblas=*=*accelerate` as described here: https://conda-forge.org/docs/maintainer/knowledge_base/#switching-blas-implementation

**Comment by jaredjeya** [comment]
Hi, I'm also seeing this issue with 'numpy=2.3.3', 'libblas=3.9.0=37_h010d69f_newaccelerate', and 'python=3.12.12' on a Macbook Air M4 - in general, for any linear algebra operation including matmuls. I'm certain I'm not using OpenBLAS so this seems odd!

**Comment by mattip** [comment]
This issue is closed. You probably meant issue #29820 which is meant to be fixed by #30102.

**Comment by seberg** [comment]
FWIW, I could see that the issue isn't confined to matrix multiplication but also for other linear algebra functions. But, until now it wasn't really reported for those (at least not for a wheel setup), so until now it seems plausible the identical issue, but not sure.

We may have to widen up the fix to other linear algebra functions, but that might also break more important warnings/errors.
